pb/terrarium/provider/provider.proto

```diff
@@ -4,6 +4,9 @@ package terrarium.provider;
 option go_package = "github.com/terrariumcloud/terrarium/pkg/terrarium/provider";
 
 service ProviderPublisher {
+  rpc UploadProviderSourceZip (stream UploadProviderSourceZipRequest) returns (Response) {}
```

Comment:
is the plan to upload source file? it does not make sense here as we don't want to build provider in terrarium. the function should clearly describe the purpose

Reply:
No, source file won't be uploaded, the plan is that the provider will be build as a part of our workflows in some runners, **this function's role is to upload the provider binary.**
I agree to your point that the function name should be more relatable and updated it accordingly.